hive-event-listeners/apiary-gluesync-listener/src/main/java/com/expediagroup/apiary/extensions/gluesync/listener/service/GlueMetadataStringCleaner.java

```diff
@@ -15,13 +15,44 @@
  */
 package com.expediagroup.apiary.extensions.gluesync.listener.service;
 
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.amazonaws.services.glue.model.Column;
+import com.amazonaws.services.glue.model.PartitionInput;
+import com.amazonaws.services.glue.model.TableInput;
+
 /**
  * Following https://docs.aws.amazon.com/glue/latest/dg/aws-glue-api-catalog-tables.html#aws-glue-api-catalog-tables-Table
  * validations
  */
 public class GlueMetadataStringCleaner {
 
-  public String shortTo254Chars(String input) {
+  public TableInput cleanTable(TableInput input) {
+    // Clean SerDes
+    cleanColumns(input.getStorageDescriptor().getColumns());
+    // Clean Partition Keys
+    cleanColumns(input.getPartitionKeys());
+    return input;
+  }
+
+  public PartitionInput cleanPartition(PartitionInput input) {
+    // Clean SerDes
+    cleanColumns(input.getStorageDescriptor().getColumns());
+    // Clean Partition Keys
+    List<String> cleanedKeys = input.getValues().stream().map(this::removeNonUnicodeChars).collect(Collectors.toList());
+    input.setValues(cleanedKeys);
+    return input;
+  }
+
+  private void cleanColumns(List<Column> columns) {
+    for (Column column : columns) {
+      column.setName(removeNonUnicodeChars(column.getName()));
```

Comment:
Do you have an example of this happening it seems weird that this can actually happen, I'd expect hive to already fail. If we have unicode in the name of a column I'm not sure we should clean it up I think it's better to have that fail.